### ISSUE TITLE
IOS: Improve HTTP response processing in PA2Client

### ISSUE THREAD

**Issue by hvge** [lime-company/powerauth-mobile-sdk#59]
Our `PA2Client` method:
```objc
- (NSURLSessionDataTask*) postToUrl:(NSURL*)absoluteUrl
                      requestObject:(id<PA2NetworkObject>)requestObject
                            headers:(NSDictionary*)headers
                responseObjectClass:(Class)responseObjectClass
                           callback:(void(^)(PA2RestResponseStatus status, id<PA2NetworkObject> response, NSError *error))callback
```
is designed to automatically create a model object (with expected class) when the response is in all ways correct. This is right, unless the 200 HTTP status code is received, but with the error JSON in the response body. For this corner case, the `PA2Error` object is created and passed to the callback as correct response `callback(PA2RestResponseStatus_OK, >>that-error<<, nil)`. 

The result is that the application will probably crash on some "object doesn't respond to selector" exception, because the callback code is not expecting `PA2Error` in that response object. 

### Recommended changes

- `PA2Response<T>` model class should **NOT** internally set `PA2Error` object into the `responseObject` property. We should add a separate property dedicated exclusively for an error. Let's call it `responseError`.

- `PA2Client` should check the `PA2Response.responseError` even if the 200 status code is received and treat this situation as an error.

- The whole error response reporting is over-engineered in my opinion. We should simplify all that classes. What I don't like:
   - `PA2ErrorResponse` is actually not a response, but just a object with a real response error object embedded. 
   - `PA2Error` is a real response received from the server but looks like an object which should be reported to the application.
   - All this is then wrapped in `NSError` in its info dictionary, under the`PA2ErrorDomain` key. That's another naming mistake. We should use a separate string for that or provide a NSError extension to access that object in more safe way.
   - Both, `PA2ErrorResponse` & `PA2Error` doesn't implement a `description` method, so basically there's no easy way how to look at reported NSError and check what's really going on.

**Comment by hvge** [comment]
I've fixed wrong HTTP response processing in 882ff35, but we still need address the problems mentioned above. It is no longer a bug, thus changing label to enhancement.

**Comment by hvge** [comment]
After a couple of months I've decided to close this issue. It's no longer a problem (due to fix above) and the rest of ideas, when implemented, will probably break an existing contracts. We will address that code cleanup in the next major generation of PowerAuth SDK, when we'll rewrite the high level codebase to swift.
